src/zenml/integrations/huggingface/materializers/huggingface_datasets_materializer.py

```diff
@@ -33,13 +42,34 @@
 DEFAULT_DATASET_DIR = "hf_datasets"
 
 
+def extract_dataset_name(input_string) -> Optional[str]:
```

Comment:
What is `input_string`? The dataset name? The dataset ID? This needs a different name.